Bottom: 7b1adeab03ad5d94d92bf2e5dc0b28e3862d8228
Top:    1725c326ab310fb2d9d7e680938f988503114851
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 14:30:15 +0000

Fix a few random places where we fail badly if the board outline polygon computes to NULL

Usually, the board outline polygon should default to being the world region, but if a
polygon computation fails, it may return NULL, and we can crash in a few places if that
happens.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 5274072..a033a4f 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1684,6 +1684,8 @@ fill_board_outline_holes (hidGC gc, const BoxType *drawn_area)
   struct outline_info info;
 
   ensure_board_outline ();
+  if (PCB->Data->outline == NULL)
+    return;
 
   memset (&polygon, 0, sizeof (polygon));
   polygon.Clipped = PCB->Data->outline;
